TelemetryProcessors/SamplingTelemetryProcessor.ts

```diff
@@ -0,0 +1,45 @@
+import ContractsModule = require("../Library/Contracts");
+import { CorrelationContext } from "../AutoCollection/CorrelationContextManager";
+
+/**
+ *  A telemetry processor that handles sampling.
+ */
+export function samplingTelemetryProcessor(envelope: ContractsModule.Contracts.Envelope, contextObjects: { correlationContext: CorrelationContext }): boolean {
+    var samplingPercentage = envelope.sampleRate; // Set for us in Client.getEnvelope
+    var isSampledIn = false;
+
+    if (samplingPercentage === null || samplingPercentage === undefined || samplingPercentage >= 100) {
+        return true;
```

Comment:
We want to skip if it was already calculated so you can chain multiple sampling processors